test/server/cards/01_SOR/events/YoureMyOnlyHope.spec.ts

```diff
@@ -0,0 +1,76 @@
+describe('You\'re My Only Hope', function() {
+    integration(function(contextRef) {
+        describe('You\'re My Only Hope\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        leader: 'hera-syndulla#spectre-two',
+                        base: { card: 'administrators-tower', damage: 21 },
+                        hand: ['youre-my-only-hope'],
+                        deck: ['atst', 'wampa'],
+                        groundArena: ['isb-agent']
+                    },
+                    player2: {
+                        spaceArena: ['black-one#scourge-of-starkiller-base']
+                    }
+                });
+            });
+
+            it('shows the top card of the deck and allow to play it', function () {
+                const { context } = contextRef;
+                const reset = () => {
+                    context.player1.moveCard(context.youreMyOnlyHope, 'hand');
+                    context.player2.passAction();
+                };
+
+                // Scenario 1: Do nothing
+                context.player1.clickCard(context.youreMyOnlyHope);
+                // TODO: we need a 'look at' prompt for secretly revealing, currently chat logs go to all players
+                expect(context.getChatLogs(1)).toContain('You\'re My Only Hope sees AT-ST');
+                expect(context.player1).toHavePassAbilityPrompt('Play AT-ST, it costs 5 less');
+
+                const deckBeforeAction = context.player1.deck;
+                context.player1.passAction();
+                expect(context.player1.deck).toEqual(deckBeforeAction);
+                expect(context.atst).not.toBeInZone('groundArena', context.player1);
```

Comment:
I think it would be easier to check that the ATST is in the deck and remove the deckbeforeAction completely